FP Fit Lit Part 1 Updates

Kalikoze opened the pull request:
### Description of Changes
- Class architecture has been removed from spec
- Rubric has been updated to M1 structure with Wow, Yes, Not Yet scoring
- Closures are added to WOW
- Spec references 2 files including `scripts.js` and `domUpdates.js`

### Commentary
I closely compared this project to the [changes added to whats-cookin](https://github.com/turingschool/front-end-curriculum/pull/646) while also trying to keep the new additions with the agile workflow.  My goal in this was to help norm across both projects.

This also goes hand in hand with the [updates made to the starter kit in this PR](https://github.com/turingschool-examples/fitlit-starter-kit/pull/411), which also needs to be reviewed.

### Questions for reviewers
* Is the **WOW** for _Testing_ accurate?  I noticed this was missing on the What's Cookin Part One project spec, so I wanted to make sure it was normed across both.
* I've updated the structure of the rubric to have drop downs similar to the [PR for Travel Tracker](https://github.com/turingschool/front-end-curriculum/pull/657).  However, this is different from What's Cookin, so I just want to confirm if this is the norm moving forward.
* There are 5 iterations for FitLit Part One, but only 3 for What's Cookin Part One.  Should this get cut down some?  Especially since Sleep, Hydration, and Activity data are all asking similar things? 
* Based on the above, the old rubric asked for all iterations plus an extension for a **4**, and all iterations for a **3**.  However, the way the spec is currently setup, there are no extensions written in.  I've added back in some of the old extensions to match this expectation, but am curious on others' thoughts.

Closes #617

Commit:
Update agile FitLit part one to remove classes and focus more on FP

Commit:
Continue to udpate fp fitlit part one moving some sections around and making small adjustments to the rubric

niksseif commented on projects/module-2/fitlit-part-one-agile.md:
```diff
@@ -255,6 +221,63 @@ You should create functions that:
+* Implement one of the listed third-party libraries or one of your choosing with instructor approval.
+* Design a step challenge between friends. Assign your user a few friends from the user data file. Add the functions you need and a display on the dashboard to see their friends step count for a whole week. The display should also show who had the most steps for that week.
+* Calculate and display what days a user had increasing steps for 3 or more days.
+* Build drag-and-drop widgets (*if you did not go with a widget theme, then add some widgets for this extension*). You can use a 3rd-party library to help you accomplish this.
```
Thanks for working on this, Travis! This looks great. Just out of curiosity, do we have a reliable widget library that we can introduce here? If not, perhaps it would be a good idea to add a note suggesting the use of widgets library with instructor approval.

Kalikoze replied:
Great point Nik, thanks for the suggestion.  After talking with Heather and Robbie, it sounds like most students are using the [HTML Drag and Drop API](https://developer.mozilla.org/en-US/docs/Web/API/HTML_Drag_and_Drop_API), so I'll add that in!

niksseif reviewed (commented):
Hey Travis! This looks great - thanks for working on it! I've left a suggestion for you to consider.

hfaerber commented on projects/module-2/fitlit-part-one-agile.md:
```diff
@@ -101,185 +105,244 @@ You should not be hardcoding the date.  You should be using code to find the las
+Instead, for your tests, you should create small, sample datasets that match the structure of the application data. By creating this sample dataset, you will also know if your functions are working correctly because you can do the calculations by hand with a much smaller dataset.
+
+**You are *expected* to test:**
+- All functions that do not update the DOM.  This means everything in your `scripts.js` file that should be tested.
```
As I worked on the testing lesson I realized some challenges that students are going to encounter with testing.  If their functions are updating a global variable, without taking it in as an argument, it's going to be really hard or impossible to test those functions.  Hopefully this will guide them towards more pure solutions but I suspect they will still end up with some impure functions that are really hard or impossible to test. 

I'm wondering what the best wording is for what we expect them to test?  "All functions that do not update the DOM."  is probably not realistic (though I used the same wording in Whats Cookin Part 1)

hfaerber replied:
I think we need to remove the word "that" from the second sentence :)

Kalikoze replied:
Hmm good point Heather, I'll definitely think about rewording that some more!  Could it be reworded to say "All pure functions that do not update the DOM?"  Or "Strive to test all functions that do not update the DOM?"

Kalikoze replied:
After discussing with Heather more, we discussed updating this to focus more on what they "should *NOT*" test including DOM updates and network requests.

hfaerber commented on projects/module-2/fitlit-part-one-agile.md:
```diff
@@ -101,185 +105,244 @@ You should not be hardcoding the date.  You should be using code to find the las
+
+## Rubric
+
+For the rubric sections below, you will be scored as **Wow**, **Yes** or **Not Yet** depending on whether you have demonstrated competency in that area. Each section lists examples of what types of things we may be looking for as demonstrations of competency. Just as there are many ways to approach code, there are many many ways to demonstate competency.  Thesee are just some examples.
```
Tiny typo towards the end of line 285 - extra 'e' in these

Kalikoze replied:
Good catch, thank you!

hfaerber reviewed (commented):
Thanks for putting this together Travis!  

My take on your questions:
1) Is the WOW for Testing accurate? I noticed this was missing on the What's Cookin Part One project spec, so I wanted to make sure it was normed across both.
- When we met with the Mod 2 team about WOWs in Mod 2, I think we settled on only referencing a specific WOW when there was a bulleted extra thing students could do (like responsive design) that was above and beyond the requirements.  This means some sections, like Testing, might not have a WOW.   We want them to strive for thorough and robust testing as an examples of competency.  If they do a pretty good job, that would be a YES.  If they do an amazing super-duper thorough job, that would be a WOW.  But we decided we don't want to spell that out for them in the rubric because we don't want them to only strive for "pretty good".

2) I've updated the structure of the rubric to have drop downs similar to the https://github.com/turingschool/front-end-curriculum/pull/657. However, this is different from What's Cookin, so I just want to confirm if this is the norm moving forward.
- I don't think we've normed on this but Cass did use dropdowns for the final project.  We can definitely add a ticket to make What Cooking Part 1 match this if we want it to be consistent across the projects.

3) There are 5 iterations for FitLit Part One, but only 3 for What's Cookin Part One. Should this get cut down some? Especially since Sleep, Hydration, and Activity data are all asking similar things?
- In my opinion, Whats Cooking is much harder (and better learning) than FitLit because FitLit has mostly display without user interactivity.  For that reason, I think its fine to have more iterations in FitLit than in Whats Cooking.  

4) Based on the above, the old rubric asked for all iterations plus an extension for a 4, and all iterations for a 3. However, the way the spec is currently setup, there are no extensions written in. I've added back in some of the old extensions to match this expectation, but am curious on others' thoughts.
- That's weird that the old spec referenced extensions but their weren't any.  Listing some old ones as options sounds good to me!

hfaerber commented on projects/module-2/fitlit-part-one-agile.md:
```diff
@@ -101,185 +105,244 @@ You should not be hardcoding the date.  You should be using code to find the las
+<section class="answer">
+### Functional Expectations
+
+💫ON TRACK💫 can look like:
```
I dont think I've even formed an opinion on this myself - this is more of a wondering....do we think adding "On Track" here could be confusing to students when we've been operating with WOW, Yes and Not Yet?   Maybe not, just something that had me double-take for a second.

Kalikoze replied:
Hmm that's a really good point!  So far, in M1 I haven't noticed any issues with students understanding it.  So I think for norming sakes, I might stick with what we've been doing, but if confusion does start to prop up, I'd be more than happy to update it to say **Yes**!

Commit:
Merge branch 'development' of github.com:turingschool/front-end-curriculum into fp-fitlit-part1

Commit:
Remove wow from testing in rubric, add recommendation for drag and drop, and fix typos

Kalikoze commented:
Thanks for the feedback Heather and Nik!  I've added in a recommendation for a drag-and-drop, removed the **WOW** from the testing part of the rubric, and fixed those typos.  Appreciate ya'll!
